.github/workflows/release.yml

```diff
@@ -0,0 +1,54 @@
+name: release
+on:
+  workflow_dispatch
+
+jobs:
+  publish-to-nuget:
+    runs-on: windows-latest
+    name: Publish nupkg to nuget
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Download all artifacts
+      uses: dawidd6/action-download-artifact@v2
+      with:
+        workflow: main.yml
+        branch: ${{ github.ref }}
+        path: ${{ github.workspace }}/Realm/packages/
+    - name: Extract release notes
+      run: |
+        $changelogContent = Get-Content -Raw -Path "./CHANGELOG.md"
+        $output_file = "./RELEASE_NOTES.md"
+        $regex = "(?sm)^(## \d{1,2}\.\d{1,2}\.\d{1,2} \(\d{4}-\d{2}-\d{2}\))(.+?)(\n## \d{1,2}\.\d{1,2}\.\d{1,2} \(\d{4}-\d{2}-\d{2}\))"
+        $changelogContent -match $regex
+        Set-Content -Path $output_file -Value $Matches[2]
+        echo "RELEASE_NOTES=$output_file" | Out-File $env:GITHUB_ENV -Encoding utf8
+    - name: Extract release name
+      run: |
+        $nupkgName = ls Realm/packages/Realm.Fody.$env:RELEASE_VERSION | select -expandproperty Name | Select-String Fody
+        $nupkgName -match "(?sm)\d{1,2}\.\d{1,2}\.\d{1,2}"
+        $version = $Matches[0]
+        echo "RELEASE_VERSION=$version" | Out-File $env:GITHUB_ENV -Encoding utf8
```

Comment:
Rather than use env variables here, I think we should use set-output

Reply:
I can do that but, what's the advantage? As a con it has longer syntax, what are the pros?

Reply:
I generally dislike env variables as a way of passing data between steps. The step outputs are designed precisely for that and we should prefer to use them.